proxy/common.go

```diff
@@ -19,6 +22,14 @@ var (
 	weaveWaitEntrypoint = []string{"/w/w"}
 )
 
+type UnmarshalWrongTypeError struct {
+	Field, Expected string
+}
+
+func (e *UnmarshalWrongTypeError) Error() string {
+	return fmt.Sprintf("Wrong type for %s field, expected %s", e.Field, e.Expected)
```

Comment:
This message will be confusing if the real issue is that the field is missing completely from the JSON.